### DISCUSSION TITLE
Modules loaded multiple times

### DISCUSSION THREAD

**Discussion by Massi-X** [LSPosed/LSPosed#1348] [Q&A] [answered]
Hi I need some clarification regarding the way Lsposed loads modules, indeed I see on multiple devices that various modules (including mine) are loaded multiple times but not fully... To clarify: my module is loaded when the launcher starts and print in the log an info line regarding the installed launcher version, but I see it gets loaded multiple times without even printing this line! I think this should not happen, obviously the scope is restricted to the launcher only

Thank yoy

**Comment by yujincheng08** [comment, marked as answer]
There are two entry points (basically three but just ignore resource hook).

One is `initZygote`, which is called as soon as Zygote forked this process. So for one process of one application, `initZygote` will be and only be called once. At that time, no classloader (except for the one from the system) is available.

The other is `handleLoadedPackage`, which is called whenever a single LoadedApk is called `getClassloader`, which means the classloader of a package is constructed. You might think there can only be only one package in one process, but this is false. There are many scenarios in which more than one packages can be loaded into one process: i) packages sharing the same UID and process name; ii) overlay packages from the system (for instance, MIUI will load a crash report overlay package for every process); iii) Calling `createPackageContext` or some other interfaces that load an external package.

I don't know what exactly your scenario is. But pay attention to:
1. Is your module loaded in different processes (see the PID)
2. Is your `handleLoadedPackage` calling multiple times with different package names

For other scenarios, you may need to submit a full log.

**Reply by Massi-X** [reply]
Yes this is it, different PIDs or packages

This was not a bug report, only a clarification indeed everything works fine. Thank you for your kindness
